reference-implementations/icrc25/relying-party/src/App.js

```diff
@@ -69,81 +66,21 @@ function App() {
   }, [activeAccount, fetchBalance])
 
   const verifyPermissionResponse = async (response, challenge) => {
-    const derPublicKey = Buffer.from(response.result.identities[0].publicKey, 'base64')
-    const secp256k1PublicKey = (publicKey) => {
-      try {
-        return {
-          curve: 'secp256k1',
-          raw: Secp256k1PublicKey.fromDer(publicKey).toRaw()
-        }
-      } catch {
-        return undefined
-      }
-    }
-    const ed25519PublicKey = (publicKey) => {
-      try {
-        return {
-          curve: 'ed25519',
-          raw: Ed25519PublicKey.fromDer(publicKey).toRaw()
-        }
-      } catch {
-        return undefined
-      }
-    }
-    const p256PublicKey = (publicKey) => {
-      const ECDSA_P256_OID = Uint8Array.from([
-        ...[0x30, 0x13], // SEQUENCE
-        ...[0x06, 0x07], // OID with 7 bytes
-        ...[0x2a, 0x86, 0x48, 0xce, 0x3d, 0x02, 0x01], // OID ECDSA
-        ...[0x06, 0x08], // OID with 8 bytes
-        ...[0x2a, 0x86, 0x48, 0xce, 0x3d, 0x03, 0x01, 0x07], // OID P-256
-      ]);
-
-      try {
-        return {
-          curve: 'p256',
-          raw: unwrapDER(publicKey, ECDSA_P256_OID)
-        }
-      } catch {
-        return undefined
-      }
-    }
-
-    const publicKey = secp256k1PublicKey(derPublicKey) ?? ed25519PublicKey(derPublicKey) ?? p256PublicKey(derPublicKey)
-    if (publicKey === undefined) {
-      throw new Error('Public key not supported')
-    }
-
-    const signatureRaw = Buffer.from(response.result.signature, 'base64')
-    const dataRaw = Buffer.concat([
-      Buffer.from(new TextEncoder().encode('\x0Aic-wallet-challenge')), 
+    const derPublicKey = Uint8Array.from(Buffer.from(response.result.identities[0].publicKey, 'base64'))
```

Comment:
`Buffer` is a special type of`Uint8Array`, you shouldn't need to explicitly wrap it with `Uint8Array.from` 🤔